crates/y-sweet/src/server.rs

```diff
@@ -510,6 +514,34 @@ async fn handle_socket_upgrade(
     Ok(ws.on_upgrade(move |socket| handle_socket(socket, awareness, cancellation_token)))
 }
 
+async fn handle_socket_upgrade_deprecated(
+    ws: WebSocketUpgrade,
+    Path(doc_id): Path<String>,
+    Query(params): Query<HandlerParams>,
+    State(server_state): State<Arc<Server>>,
+) -> Result<Response, AppError> {
+    tracing::warn!(
+        "/doc/ws/:doc_id is deprecated; call /doc/:doc_id/auth instead and use the returned URL."
```

Comment:
I wonder what a good way to detect these on our side might be

Reply:
This doesn't impact backends served on Jamsocket, since they use the `serve-doc` server instead. This change just makes the location of the `ws` endpoint for self-hosted (i.e. `y-sweet serve`) servers consistent with what we already do on Plane/Jamsocket y-sweet.